pkg/base/cockpit.js

```diff
@@ -2510,6 +2510,62 @@ function full_scope(cockpit, $) {
         return new HttpClient(endpoint, options || { });
     };
 
+    /* ---------------------------------------------------------------------
+     * Permission
+     */
+
+    var authority = null;
+
+    function Permission(options) {
+        var self = this;
+        self.allowed = null;
+
+        var user = cockpit.user;
+        var group = null;
+
+        if (options)
+            group = options.group;
+
+        function decide() {
+            var allowed = null;
```

Comment:
The logic in this function is hard to follow and hard to extend later.  Would the following be better?

```
function decide() {
  if (user.id === undefined)
     return null;

  if (user.id === 0)
     return true;

  if (group && user.groups) {
     ...
     if (name == group)
       return true;
  }

  return false;
}

function user_changed() {
  var allowed = decide();
  if (self.allowed !== allowed) {
    self.allowed = allowed;
    $(self).triggerHandler("changed");
  }
}
```